portfolio/src/main/java/com/google/sps/servlets/NewCommentDataServlet.java

```diff
@@ -63,4 +81,50 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   private boolean verified(String user, String content) {
     return !user.isEmpty() && !content.isEmpty();
   }
+
+
+  private String getBlobKey(HttpServletRequest request){
```

Comment:
This method is named "getBlobKey", which makes me think that it has no [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)), but it actually uploads and deletes images.

Usually getter methods shouldn't do anything besides getting, so I think you can find a better name for this.